### DISCUSSION TITLE
How do I end a chat session from a ConversableAgent instance's reply_func

### DISCUSSION THREAD

**Discussion by mark-at-hypothetic** [microsoft/autogen#424] [Q&A]
I'd like to end a chat from a reply_func, 

```
    ....
class IntentAgnet(autogen.AssistantAgent):
    def __init__(...):
       ...
      self.register_reply(
        trigger=autogen.UserProxyAgent,
        reply_func=IntentAgent._generate_reply_user)

    def _generate_reply_user(self):
        message = messages[-1]
        reply = self.generate_reply([message])

        if reply == "validation_ok":
            # it is validated. Now end the chat
            TERMINATE?
```

How do I terminate from here? Or is that bad design? I tried setting the `self.update_max_consecutive_auto_reply(0)` , but then this agent still sends a reply over to the other agent.

**Comment by sonichi** [comment]
Solution is simple: `return True, None`

**Reply by mark-at-hypothetic** [reply]
Nice! It partly works though, when I do that, the other agent gets to reply one more time before it terminates. How do I skip that as well?

**Reply by sonichi** [reply]
return True, "TERMINATE"

**Reply by ip-aka-ilamparithi** [reply]
@sonichi - what does return True means here? . Sorry couldn't find anything in docs.
